serde_derive/src/internals/ast.rs

```diff
@@ -66,7 +102,10 @@ impl<'a> Container<'a> {
         let mut attrs = attr::Container::from_ast(cx, item);
 
         let mut data = match &item.data {
-            syn::Data::Enum(data) => Data::Enum(enum_from_ast(cx, &data.variants, attrs.default())),
+            syn::Data::Enum(data) => Data::Enum(
+                VariantMix::Any,
+                enum_from_ast(cx, &data.variants, attrs.default()),
+            ),
```

Comment:
The code below this is only applying rename_all rules. That does not change the kind of a variant tag. The correct `VariantMix` can be set right here. We do not need to use `Any` as a placeholder.

Alternatively it may be more convenient to not even store `VariantMix` and avoid needing all those `_` placeholders throughout the rest of serde_derive. After checking has happened, a correct `VariantMix` can be computed efficiently from just the Vec\<Variant\> by looking at only the first variant's name.